modules/docs/src/04-codegen/01-customisation.md

```diff
@@ -325,3 +325,92 @@ string Email
 By default, smithy4s renders collection types as unwrapped EXCEPT when the collection has been refined. In this case, the collection will be rendered within a newtype by default. If you wish your refined collection be rendered unwrapped, you can accomplish this using the same `@unwrap` trait annotation on it.
 
 :::
+
+## Default rendering
+
+Smithy4s allows you to customize how defaults on the fields of smithy structures are rendered inside of case classes. There are three options:
+
+- `FULL`
+- `OPTION_ONLY`
+- `NONE`
+
+The default is `FULL`.
+
+#### FULL
+
+`FULL` means that default values are rendered for all field types. For example:
+
+```kotlin
+use smithy4s.meta#defaultRender
+
+@defaultRender(mode: "FULL")
```

Comment:
not quite sure it should be a trait. The preference probably applies to all case classes that users generate, so I'm thinking it should be a piece of metadata instead.

Reply:
two things:
1.  >preference probably applies to all case classes

The opposite could be just as true I guess, some people may want to change the behaviour for one specific case class.

2. >  the rendering of trait values relies on the ordering of parameters

I don't understand what you mean by the `rendering of trait values` in this context, do you have an example?

Reply:
1) we have optimise user-facing features in terms of what most users will benefit from. Though you are not technically incorrect, I'm pretty sure that if someone dislikes default values in one case, they'll dislike default values in all cases, because default values have very tangible downsides to them. Therefore, it's likely that a piece of UX that would allow to disable defaults altogether across the whole code-gen pipeline, people who dislike defaults will be happy about the low-boilerplate solution. 
2) I meant rendering of hints.

Reply:
Good points, I forgot that we had talked about using the metadata for this. I'll update the PR with that and using named parameters in hints 👍

Reply:
Done in: https://github.com/disneystreaming/smithy4s/pull/414/commits/0b9d512413465421fc8d1c32403adf1adf85a1e0

lmk if this doesn't match what you are talking about wrt the named parameters in hints

Reply:
> lmk if this doesn't match what you are talking about wrt the named parameters in hints 

It is indeed